test/e2e/framework/kcp.go

```diff
@@ -213,12 +231,34 @@ type RunningServer interface {
 	Artifact(t *testing.T, producer func() (runtime.Object, error))
 }
 
+// KcpConfigOption a function that wish to modify a given kcp configuration
+type KcpConfigOption func(*kcpConfig) *kcpConfig
+
+// WithScratchDirectories adds custom scratch directories to a kcp configuration
+func WithScratchDirectories(artifactDir, dataDir string) KcpConfigOption {
+	return func(cfg *kcpConfig) *kcpConfig {
+		cfg.ArtifactDir = artifactDir
+		cfg.DataDir = dataDir
+		return cfg
+	}
+}
```

Comment:
why isn't this the default to have independent artifact and data dirs?

Reply:
The scratch dirs are provided/generated by default by the test framework. See https://github.com/kcp-dev/kcp/pull/2197/files#diff-6119c8a3bef16c15807ae827fde6e0f6b9e6d893969064b4cc24063755a90fffR106 and https://github.com/kcp-dev/kcp/pull/2197/files#diff-6119c8a3bef16c15807ae827fde6e0f6b9e6d893969064b4cc24063755a90fffR141

Reply:
The `PrivateKcpServer` was extended so that the scratch dirs can be provided externally by the test.

Reply:
if they are created already, is the problem that you cannot access them?

Reply:
I wanted to create these files before starting my tests, so that the cache-server artifact, data are placed in the same location.

Reply:
I cannot rely on the `PrivateKcpServer` method because it immediately starts the kcp server.

Reply:
and in a standalone mode the kcp requires a path to kubeconifig that holds config for the cache-server